src/core/renderer_vk/renderer_vk.cpp

```diff
@@ -0,0 +1,546 @@
+	presentInfo.setSwapchains(swapchain.get());
+	presentInfo.setImageIndices(swapchainImageIndex);
+
+	if (const auto presentResult = presentQueue.presentKHR(presentInfo); presentResult == vk::Result::eSuccess) {
+	} else {
+		switch (presentResult) {
+			case vk::Result::eSuboptimalKHR:
+			case vk::Result::eErrorOutOfDateKHR: {
+				// Surface resized
+				vk::Extent2D swapchainExtent;
+				{
+					int windowWidth, windowHeight;
+					SDL_Vulkan_GetDrawableSize(targetWindow, &windowWidth, &windowHeight);
+					swapchainExtent.width = windowWidth;
+					swapchainExtent.height = windowHeight;
+				}
+				recreateSwapchain(surface.get(), swapchainExtent);
+				break;
+			}
+			default: {
+				Helpers::panic("Error presenting swapchain image: %s\n", vk::to_string(presentResult).c_str());
+			}
+		}
+	}
+
+	currentFrame = ((currentFrame + 1) % swapchainImageCount);
+}
+
+void RendererVK::initGraphicsContext(SDL_Window* window) {
+	targetWindow = window;
+	// Resolve all instance function pointers
+	static vk::DynamicLoader dl;
+	VULKAN_HPP_DEFAULT_DISPATCHER.init(dl.getProcAddress<PFN_vkGetInstanceProcAddr>("vkGetInstanceProcAddr"));
+
+	// Create Instance
+	vk::ApplicationInfo applicationInfo = {};
+	applicationInfo.apiVersion = VK_API_VERSION_1_1;
+
+	applicationInfo.pEngineName = "Alber";
+	applicationInfo.engineVersion = VK_MAKE_VERSION(1, 0, 0);
+
+	applicationInfo.pApplicationName = "Alber";
+	applicationInfo.applicationVersion = VK_MAKE_VERSION(1, 0, 0);
+
+	vk::InstanceCreateInfo instanceInfo = {};
+
+	instanceInfo.pApplicationInfo = &applicationInfo;
+
+	std::vector<const char*> instanceExtensions = {
+#if defined(__APPLE__)
+		VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME,
+#endif
+		VK_EXT_DEBUG_UTILS_EXTENSION_NAME,
+	};
+
+	// Get any additional extensions that SDL wants as well
+	{
+		unsigned int extensionCount = 0;
+		SDL_Vulkan_GetInstanceExtensions(window, &extensionCount, nullptr);
+		std::vector<const char*> sdlInstanceExtensions(extensionCount);
+		SDL_Vulkan_GetInstanceExtensions(window, &extensionCount, sdlInstanceExtensions.data());
+
+		instanceExtensions.insert(instanceExtensions.end(), sdlInstanceExtensions.begin(), sdlInstanceExtensions.end());
+	}
+
+#if defined(__APPLE__)
+	instanceInfo.flags |= vk::InstanceCreateFlagBits::eEnumeratePortabilityKHR;
+#endif
+
+	instanceInfo.ppEnabledExtensionNames = instanceExtensions.data();
+	instanceInfo.enabledExtensionCount = instanceExtensions.size();
+
+	if (auto createResult = vk::createInstanceUnique(instanceInfo); createResult.result == vk::Result::eSuccess) {
+		instance = std::move(createResult.value);
+	} else {
+		Helpers::panic("Error creating Vulkan instance: %s\n", vk::to_string(createResult.result).c_str());
+	}
+	// Initialize instance-specific function pointers
+	VULKAN_HPP_DEFAULT_DISPATCHER.init(instance.get());
+
+	// Enable debug messenger if the instance was able to be created with debug_utils
```

Comment:
Vk and OpenGL debug messages should also maybe have their own CMake flag eventually but this should def stay enabled for Vk at the moment

Reply:
I def agree, this is what the driver will use to for its own diagnostic messaging as well. The NVidia driver is known to push messages through here at times in some cases as well during some crashes or exceptional moments.